backend/app/api/endpoints/mcp_providers.py

```
@@ -37,6 +42,56 @@ async def list_mcp_providers(
+        raw_tools = await asyncio.wait_for(_connect_and_get_tools(config), timeout=15)
+        tools = [
+            MCPToolInfo(
+                name=getattr(tool, "name", ""),
+                description=getattr(tool, "description", ""),
+            )
+            for tool in raw_tools
+        ]
+        return MCPTestResponse(success=True, tools=tools)
+    except asyncio.TimeoutError:
+        return MCPTestResponse(success=False, error="Connection timed out (15s)")
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)